src/compiler/checker.ts

```diff
@@ -7163,17 +7163,49 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
             }
 
-            const modifiers = isReadonlySymbol(propertySymbol) ? [factory.createToken(SyntaxKind.ReadonlyKeyword)] : undefined;
-            if (modifiers) {
-                context.approximateLength += 9;
+            if (propertySymbol.flags & SymbolFlags.Accessor) {
+                if (propertySymbol.flags & SymbolFlags.GetAccessor) {
+                    const getAccessorDecl = find(propertySymbol.declarations, decl => decl.kind === SyntaxKind.GetAccessor) as GetAccessorDeclaration;
+                    const getAccessorSignature = factory.createGetAccessorDeclaration(
+                        /*modifiers*/ undefined,
+                        propertyName,
+                        [],
+                        propertyTypeNode,
+                        /*body*/ undefined);
+                    setCommentRange(getAccessorSignature, getAccessorDecl);
+                    typeElements.push(getAccessorSignature);
+                }
+                if (propertySymbol.flags & SymbolFlags.SetAccessor) {
+                    const setAccessorDecl = find(propertySymbol.declarations, decl => decl.kind === SyntaxKind.SetAccessor) as SetAccessorDeclaration;
+                    const parameterName = setAccessorDecl?.parameters?.length > 0 ? setAccessorDecl.parameters[0].name : "arg";
+                    const setAccessorSignature = factory.createSetAccessorDeclaration(
+                        /*modifiers*/ undefined,
+                        propertyName,
+                        [factory.createParameterDeclaration(
+                            /*modifiers*/ undefined,
+                            /*dotDotDotToken*/ undefined,
+                            parameterName,
+                            /*questionToken*/ undefined,
+                            propertyTypeNode

```

Comment:
Rather than reusing the `propertyTypeNode`, we'd need to at least (deep) clone it - potentially putting the same node in the tree at two places can have funky results sometimes. But, better yet, would be fixing the differing get/set type bug while we're here! You can get the `set`er type by using `getWriteTypeOfSymbol` on the `propertySymbol`, which you can then feed through `serializeTypeForDeclaration`, same as is done for the reading type. You might need to duplicate the `getNonMissingTypeOfSymbol` into a `getNonMissingWriteTypeOfSymbol` to handle the `missing` type (the special `undefined`-like type we represent non-existent properties with in some compiler settings) correctly, though.

Reply:
Makes sense, @weswigham. I should be able to implement it this weekend.

Reply:
This is done, and I think the updated baselines demonstrate it's working. I wasn't sure if any of the `createElidedInformationPlaceholder` or `propertyIsReverseMapped` stuff applies to the setter type. If the answer is "yes" (or even "maybe"), let me know and I'll dig deeper. Thanks for the ongoing review!